src/datasets/dataset_dict.py

```diff
@@ -1417,6 +1423,57 @@ def push_to_hub(
             revision=branch,
         )
 
+    def to_pandas(
+        self,
+        splits: Optional[Union[Literal["all"], List[str]]] = None,
```

Comment:
We could also add our `Split` to the type hint.
```python
df_all = dataset_dict.to_pandas(splits=Split.ALL)
```

Reply:
Should we also allow `str`?
```python
df_test = dataset_dict.to_pandas(splits="test")
```

Reply:
If one wants to choose one split they already do `dataset_dict["test"].to_pandas()` - I don't think that introducing `splits="test"` would make it particularly easier.

Although since we don't support the `Split` API fully (e.g. doing `"train+test[:20]"`) I wouldn't necessarily add `Split` in the type hint